pluto.jl

```diff
@@ -126,26 +126,40 @@ $$[CPU] ↔ [RAM] ↔ [硬盘缓存] ↔ [硬盘]$$
+在该尺度上，从 RAM 中读取数据需要花费 500 个时钟周期。
+通过将数据拷贝到 RAM 可以减缓硬盘延迟。与之类似，RAM 上的数据也可以拷贝到 CPU 芯片上的一小块内存，称作**缓存（cache）**。
+缓存更快的原因是，它物理上位于CPU芯片（减少线路延迟），同时它使用一类更快的存储方式，静态RAM，而不是主流 RAM 所使用的动态 RAM方案。
+因为缓存必须位于 CPU 上所带来的尺寸限制，以及由于更加昂贵的生产成本，所以经典 CPU 缓存大约仅包含  $10^8$ 比特， 这仅是 RAM 容量的 1/1000。
+实际上，CPU 存在多级缓存，但此处仅简化为一级 **缓存** ： 
 
-$$[CPU] ↔ [CPU CACHE] ↔ [RAM] ↔ [DISK CACHE] ↔ [DISK]$$
+$$[CPU] ↔ [CPU 缓存] ↔ [RAM] ↔ [硬盘缓存] ↔ [硬盘]$$
 
-When the CPU requests a piece of data from the RAM, say a single byte, it will first check if the memory is already in cache. If so, it will read it from there. This is much faster, usually just one or a few clock cycles, than access to RAM. If not, we have a *cache miss*, where your program will stall for around 100 nanoseconds while your computer copies data from RAM into the cache.
+当 CPU 从 RAM 请求一段数据时，比如一个字节，会首先检查数据所处的内存是否已经在 CPU 缓存中。
+如果是，那么将从 CPU 缓存读取。这相比从 RAM 访问更快，通常仅需要几个时钟周期。
+如果否，那么将得到**缓存未命中**，你的程序会等待约 100 ns 直到计算机将数据从 RAM 拷贝到 CPU 缓存。
 
-It is not possible, except in very low-level languages, to manually manage the CPU cache. Instead, you must make sure to use your cache effectively.
+除了非常低层级的语言外，手动管理 CPU 缓存是不可能的。相反地，您必须确保有效使用缓存。
 
-First, you strive to use as little memory as possible. With less memory, it is more likely that your data will be in cache when the CPU needs it. Remember, a CPU can do approximately 500 small operations in the time wasted by a single cache miss.
+首先，你争取使用尽可能小的内存。使用的内存越小，当 CPU 需要数据时，它就越有可能在缓存上。 
+请记住，在一次缓存未命中所浪费的时间内，CPU 可以执行约 500 次小操作。
 
-Effective use of the cache comes down to *locality*, temporal and spacial locality:
-* By *temporal locality*, I mean that data you recently accessed likely resides in cache already. Therefore, if you must access a piece of memory multiple times, make sure you do it close together in time.
-* By *spacial locality*, I mean that you should access data from memory addresses close to each other. Your CPU does not copy *just* the requested bytes to cache. Instead, your CPU will always copy data in larger chunks called *cache lines* (usually 512 consecutive bits, depending on the CPU model).
+缓存的有效使用可以总结为**局域性**，时间和空间局域性：
+* **时间局域性**，最近可能访问的数据都应该在缓存中。因此，如果需要多次访问一小块内存，请确保在紧挨着的时间内执行。
+* **空间局域性**，数据的访问地址应该彼此接近。CPU不会**仅仅**把请求的字节拷贝到内存中。
+相反地，CPU会以被称作**缓存线（cache line）**的块尺寸拷贝数据（通常是 512个连续比特，具体取决于 CPU 型号）。
 
-To illustrate this, let's compare the performance of the `random_access` function above when it's run on a short (8 KiB) vector, compared to a long (16 MiB) one. The first one is small enough that after just a few accessions, all the data has been copied to cache. The second is so large that new indexing causes cache misses most of the time. 
+为了说明上述差别，让我们比较两种情况下 `random_access` 函数的性能：第一种访问 8 KiB 大小的短向量，第二种访问 16 MiB 大小的长向量。
+前者中的向量足够小，因此当几次访问过后数据就都被拷贝到缓存上。后者中的向量如此之大，以至于新的索引大多会触发缓存未命中。
```

Comment:
好像很少会说 “触发缓存未命中” 这种话，一般更顺畅的表达可能是“触发缓存”以及“避免缓存未命中”

```suggestion
在前者短向量的情况下，因为数据比较小所以几次访问过后全部数据都被存到 CPU 缓存中了。在后者长向量的情况下，因为数据比较大，进行新的索引时缓存几乎都没有命中。
```